tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -2643,6 +2681,68 @@ HRESULT CorProfiler::RewriteForTelemetry(const ModuleMetadata& module_metadata,
     return hr;
 }
 
+HRESULT CorProfiler::RewriteIsManualInstrumentationOnly(const ModuleMetadata& module_metadata, ModuleID module_id)
+{
+    //
+    // *** Get Instrumentation TypeDef
+    //
+    mdTypeDef instrumentationTypeDef;
+    HRESULT hr = module_metadata.metadata_import->FindTypeDefByName(instrumentation_type_name.c_str(),
+                                                                    mdTokenNil, &instrumentationTypeDef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error rewriting IsManualInstrumentationOnly on getting Instrumentation TypeDef");
+        return hr;
+    }
+
+    //
+    // *** IsManualInstrumentationOnly MethodDef ***
+    //
+    constexpr COR_SIGNATURE isAutoEnabledSignature[] = {IMAGE_CEE_CS_CALLCONV_DEFAULT, 0, ELEMENT_TYPE_BOOLEAN};
+    mdMethodDef isAutoEnabledMethodDef;
+    hr = module_metadata.metadata_import->FindMethod(instrumentationTypeDef, WStr("IsManualInstrumentationOnly"),
+                                                     isAutoEnabledSignature, 3, &isAutoEnabledMethodDef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error rewriting IsManualInstrumentationOnly on getting IsManualInstrumentationOnly MethodDef");
+        return hr;
+    }
+
+    ILRewriter methodRewriter(this->info_, nullptr, module_id, isAutoEnabledMethodDef);
+    methodRewriter.InitializeTiny();
+
+    // Modify method from this:
+    // IL_0000: ldc.i4.1
+    // IL_0001: ret
+    //
+    // to this:
+    // IL_0000: ldc.i4.0
+    // IL_0001: ret
+    ILRewriterWrapper wrapper(&methodRewriter);
```

Comment:
NIT. I'd look for the instruction, rather that blindly trusting the `ldc.i4.1` is going to be the first one

Reply:
Maybe I've misunderstood, but I thought that I was specifying the exact commands here rather than anything else? i.e. it doesn't _matter_ what was there before does it, I'm just specifying the "new" body I thought?